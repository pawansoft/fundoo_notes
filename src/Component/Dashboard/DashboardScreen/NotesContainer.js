import React, { Component } from 'react';
import {
    View,
    ScrollView,
} from 'react-native'
import { Card, Paragraph, Title } from 'react-native-paper'
import NotesContainerStyle from '../../../Style/NotesContainerStyle';
import SQLiteCRUDService from '../../../../Services/SQLite_service/SQLiteCRUDService';

export default class NotesContainer extends Component {
    constructor(props) {
        super(props)
        this.state = {
            notesFromSQLite: []
        }
    }

    async componentDidMount() {
        await SQLiteCRUDService.getDetailsFromSQLiteDatabase()
            .then(async (data) => {
                var temp = []
                if (data.rows.length != 0) {
                    for (let i = 0; i < data.rows.length; ++i)
                        temp.push(data.rows.item(i));
                    await this.setState({
                        notesFromSQLite: temp
                    })
                }
            })
    }

    updateNote = (key, title, note) => {
        this.props.navigation.push('NewNotes',
            { key: key, title: title, note: note })
    }

    render() {
        return (
            <View>
                <ScrollView>
                    <View style={{ flexDirection: 'row', flexWrap: 'wrap' }}>
                        {this.state.notesFromSQLite.reverse().map(val => (                   
                            <React.Fragment key={val.NoteKey}>
                                {console.log(val.Notes)}
                                {val.isDeleted == "false" && val.isArchive == "false"?
                                    (
                                        <Card
                                            onPress={() => this.updateNote(val.NoteKey, val.Title, val.Notes)}
                                            style={(this.props.listview) ? NotesContainerStyle.container_list : NotesContainerStyle.container}>
                                            <Card.Content style={{ backgroundColor: 'white' }}>
                                                <Title style={{ color: 'black' }}>
                                                    {val.Title}
                                                </Title>
                                                <Paragraph style={{ color: 'black' }}>
                                                    {val.Notes}
                                                </Paragraph>
                                            </Card.Content>
                                        </Card>)
                                    : null}
                            </React.Fragment>
                        ))
                        }
                    </View>
                </ScrollView>
            </View>
        )
    }
}

