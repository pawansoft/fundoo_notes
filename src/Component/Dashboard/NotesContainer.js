import React,{ Component } from 'react';
import {
    View,
    Text, 
    TouchableOpacity,
    ScrollView,
} from 'react-native'
import { strings } from '../../Localization/Localization';
import RegisterStyle from '../../Style/Register';
import AsyncStorage from '@react-native-async-storage/async-storage';
import FirebaseService from '../../../Services/firebase_services/NoteServices';
import {Card, Paragraph, Title} from 'react-native-paper'
import NotesContainerStyle from '../../Style/NotesContainerStyle';

export default class NotesContainer extends Component{
    constructor(props){
        super(props)
        this.state = {
            notes : []
        }
    }

    async componentDidMount(){
        const userid = await AsyncStorage.getItem('userId');
        await FirebaseService._getNoteService(userid).then(async data => {
            let notes = data ? data: {}
            await this.setState({
                notes : notes
            })
        })
    }

    SigninInsteadNavigationHandler = () => {
        this.props.navigation.navigate('Login')
    }

    goToResponsiveWebsiteNavigationHandle = () => {
        this.props.navigation.navigate('ResponsiveImg')
    }

    updateNote = (key) => {
        this.props.navigation.push('NewNotes', { key : key, notes : this.state.notes[key]})
    }
    
    render(){
        let noteKey = Object.keys(this.state.notes);
        return(
            <View>               
                <ScrollView>
                    <View style = {{flexDirection : 'row', flexWrap : 'wrap'}}>
                        {noteKey.length > 0 ?
                            noteKey.reverse().map(key => (
                                <Card 
                                style = {(this.props.listview)? NotesContainerStyle.container_list : NotesContainerStyle.container}
                                key = {key}
                                onPress = {() => this.updateNote(key)}>   
                                  <Card.Content>
                                      <Title style = {NotesContainerStyle.container_title}>
                                          {this.state.notes[key].NotesDetail.title}
                                      </Title>
                
                                      <Paragraph>
                                          {this.state.notes[key].NotesDetail.title}
                                      </Paragraph>
                                     

                                  </Card.Content>  
                                </Card>
                            ))
                            
                            :null
                        }

                    </View>
                </ScrollView>
                <TouchableOpacity style = {RegisterStyle.touchable_opacity_style}
                    onPress = {this.SigninInsteadNavigationHandler}>
                        <Text style ={RegisterStyle.button_text}> 
                            {strings.signin}
                        </Text>
                </TouchableOpacity>
            </View>
        )
    }
}