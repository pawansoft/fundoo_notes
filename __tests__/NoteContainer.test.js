import React from 'react';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'
import NotesContainer from '../src/Component/Dashboard/NotesContainer';
import MockAsyncStorage from 'mock-async-storage';

const mockImpl = new MockAsyncStorage();
jest.mock('@react-native-async-storage/async-storage', () => mockImpl);

jest.mock('react-native-fetch-blob', () => {
  return {
    DocumentDir: () => {},
    polyfill: () => {},
  }
});

configure({adapter: new Adapter()})

jest.mock('react-native-localization', () => class RNLocalization {
    language = 'en-US'
  
    constructor (props) {
      this.props = props
      this.setLanguage(this.language)
    }
  
    setLanguage (interfaceLanguage) {
      this.language = interfaceLanguage
      if (this.props[interfaceLanguage]) {
        var localizedStrings = this.props[this.language]
        for (var key in localizedStrings) {
          if (localizedStrings.hasOwnProperty(key)) this[key] = localizedStrings[key]
        }
      }
    }
})
describe('Test Note Container', () => {
    it('ProvideNoteContainer_WhenSnapMatches_TestShouldPass', () => {
        const component = shallow(<NotesContainer/>);
        expect(component).toMatchSnapshot();
    })

    it('whenReadingNotesFromDatabase_ShouldStoreAtTheStateNotes', async () => {
        const component = shallow(<NotesContainer/>);
        await component.instance().componentDidMount();
        expect(component.instance().state.notes).not.toBe(null)
    })
})