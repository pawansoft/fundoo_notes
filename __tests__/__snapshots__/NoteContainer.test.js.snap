// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Note Container ProvideNoteContainer_WhenSnapMatches_TestShouldPass 1`] = `ShallowWrapper {}`;
